packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -264,13 +280,43 @@ export default async function processDeployment({
 
       // Handle alias-assigned event
       if (event.type === 'alias-assigned') {
-        output.stopSpinner();
+        stopSpinner();
         event.payload.indications = indications;
         return event.payload;
       }
     }
   } catch (err) {
-    output.stopSpinner();
+    stopSpinner();
     throw err;
   }
 }
+
+export function displayBuildLogs(
+  client: Client,
+  deployment: Deployment,
+  follow?: true
+): AbortController;
+export function displayBuildLogs(
+  client: Client,
+  deployment: Deployment,
+  follow: false
+): Promise<void>;
+export function displayBuildLogs(
+  client: Client,
+  deployment: Deployment,
+  follow: boolean = true
+) {
+  const abortController = new AbortController();
+  const promise = printEvents(
+    client,
+    deployment.id,
+    {
+      mode: 'logs',
+      onEvent: (event: any) => printLogShort(event, client),
+      quiet: false,
+      findOpts: { direction: 'forward', follow },
+    },
+    abortController
+  );
+  return follow ? abortController : promise;
```

Comment:
Here we might keep an "unhandled promise" if `follow=true` and the promise throws. We should handle the promise throwing if we don't return it.

Reply:
I agree with you. However, these promises are never catch by the caller function, and will interrupt the program anyway.

this `displayBuildLogs()` function already handles retries internally. I'll only reject its promise in case it retried enough, which means to stop the entire program.